Kernel/Net/LocalSocket.cpp

```diff
@@ -399,14 +399,23 @@ ErrorOr<void> LocalSocket::getsockopt(OpenFileDescription& description, int leve
 
     MutexLocker locker(mutex());
 
+    int buffer_size;
     socklen_t size;
     TRY(copy_from_user(&size, value_size.unsafe_userspace_ptr()));
 
     switch (option) {
     case SO_SNDBUF:
-        return ENOTSUP;
+        buffer_size = sizeof(*m_for_server.leak_ptr());
+        TRY(copy_to_user(static_ptr_cast<int*>(value), &buffer_size));
```

Comment:
static_ptr_cast is only for smart pointers (and in fact, only for RefPtr currently). Additionally, you are not checking that the user allocated us enough space for a size_t.